aptos-move/framework/src/natives/aggregator_natives/aggregator_v2.rs

```diff
@@ -129,16 +127,30 @@ fn native_try_add(
     mut args: VecDeque<Value>,
 ) -> SafeNativeResult<SmallVec<[Value; 1]>> {
     debug_assert_eq!(args.len(), 2);
-
     context.charge(AGGREGATOR_V2_TRY_ADD_BASE)?;
-    let aggregator_context = context.extensions().get::<NativeAggregatorContext>();
-    let mut aggregator_data = aggregator_context.aggregator_data.borrow_mut();
-    let value = pop_value_by_type(&ty_args[0], &mut args)?;
-    let (id, limit) = pop_aggregator_fields_by_type(&ty_args[0], &mut args)?;
-    let aggregator = aggregator_data.get_aggregator(id, limit)?;
-    Ok(smallvec![Value::bool(
-        aggregator.try_add(aggregator_context.resolver, value)?
-    )])
+
+    let input = pop_value_by_type(&ty_args[0], &mut args)?;
+    let agg_struct = safely_pop_arg!(args, StructRef);
+    let (agg_value, agg_max_value) = get_aggregator_fields_u128(&agg_struct)?;
```

Comment:
Why is `pop_aggregator_fields_by_type` replaced by `get_aggregator_fields_u128` everywhere? 
We should pop either `u128` values of `u64` values depending on the input type. But `get_aggregator_fields_u128` will only pop out `u128` values independent of the input type, which is not accurate.

Reply:
I mean even previous function was returning SafeNativeResult<(AggregatorID, u128)>, i.e. returning u128 for limit always - which is always OK - as it just widens the type. 
new function just returns u128 for value as well, because if execution is disabled - value can be u128, and cannot be safely cast to u64 to be put inside AggregatorID

Reply:
The older function pops either u64 or u128, and typecasts it to u128, and returns the u128. The `get_aggregator_fields_u128` function pops u128 even if the type is u64.

Reply:
hm, I got confused with what which function does, you are right, updating